### ISSUE TITLE
Connection with Socks5 Proxy

### ISSUE THREAD

**Issue by a19930905a** [spring-projects/spring-data-mongodb#4526]
Dear spring-data-mongodb , 

Just curious after I saw the Mongodb team just implement Socks5 Proxy in java driver : https://jira.mongodb.org/browse/JAVA-4347

is that means spring-data-mongodb will include this feature in next upgrade? or it still need some implementation

**Comment by mp911de** [comment]
Ideally, connectivity is provided by your infrastructure layer as we like to get a fully configured `MongoClient`. We have factory beans to configure the client. If you use the connection string to configure `MongoClient`, then you can use that feature already.

**Comment by a19930905a** [comment]
cool! thx for the answer , big appreciate

**Comment by a19930905a** [comment]
Just want to double check is I correct or not
SInce the feature is start on **mongodb-driver-sync** 4.11 version

So basically I need to ensure that my **spring-boot-starter-data-mongodb**'s **mongodb-driver-sync** have to be 4.11 version , is that correct?

**Comment by mp911de** [comment]
Yes, exactly. If you increment the version, then you're good to go.

**Comment by rishiraj88** [comment]
Thanks, @a19930905a , for the helpful query.
And, thanks to @mp911de for supporting statements.

**Comment by mp911de** [comment]
Closing this issue as we prefer e.g. Spring Boot to provide an appropriately configured `MongoClient` instance.
